StackExchange.Redis/StackExchange/Redis/RawResult.cs

```diff
@@ -117,81 +122,93 @@ internal RedisValue AsRedisValue()
             throw new InvalidCastException("Cannot convert to RedisValue: " + Type);
         }
 
+        public void Recycle(int limit = -1)
+        {
+            var arr = _itemsOversized;
+            if (limit < 0) limit = _itemsCount;
+            if (arr != null)
+            {
+                for (int i = 0; i < limit; i++)
+                {
+                    arr[i].Recycle();
+                }
+            }
+            if(_itemsOversized != null)
+                ArrayPool<RawResult>.Shared.Return(_itemsOversized, clearArray: false);
```

Comment:
Can move this into the `if` above (same with the limit check) - can avoid the dupe `if` check and the `limit` guard for the null case.

Reply:
good spot, fixed - ta